k8s/scripts/coredns_patch.sh

```diff
@@ -45,11 +47,17 @@ else
     cat $PATCH
     echo '=== end patched Corefile listing ==='
 
-    kubectl get cm coredns -n kube-system -o json | jq --arg replace "`cat $PATCH`" '.data.Corefile = $replace' | kubectl apply -f -
+    jq --arg replace "`cat $PATCH`" '.data.Corefile = $replace' $COREDNS_CM | kubectl apply -f -
 fi
 echo 'Running kubectl rollout restart for coredns...'
 kubectl rollout restart deployment/coredns -n kube-system
 echo 'Verifying that DNS resolution works inside the cluster'
```

Comment:
is the assumption here that the rollout will always succeed and there is no need to check the rollout status and have it as a precondition to run the resolution test?
Also, how do we isolate between coredns being in the transient state of rollout (restarted) vs actual dns resolution error?

Reply:
Added the rollout status check